layouts/default.vue

```diff
@@ -1,3 +1,34 @@
+<script setup lang="ts">
+type Theme = 'light' | 'dark'
+
+interface Message {
+  type: 'theme'
+  theme: Theme
+}
+
+function setTheme(theme: Theme) {
+  document.documentElement.classList.remove('light', 'dark')
+  document.documentElement.classList.add(theme)
+}
+
+const { initialTheme } = useRoute().query
+
+onMounted(() => {
+  if (initialTheme && (initialTheme === 'light' || initialTheme === 'dark')) {
+    setTheme(initialTheme)
+  }
+
+  window.addEventListener('message', (event: MessageEvent<Message>) => {
+    switch (event.data.type) {
+      case 'theme':
+        setTheme(event.data.theme)
+        break
+      default:
```

Comment:
weird not having a default case implementation, maybe use `if` instead?

Reply:
The idea was that we'll probably want to handle more events in the future, but I guess for now it can be changed to a if statement.